### PR TITLE
Archive functions

### PR DESCRIPTION
**Pull request by dusariudaykumar**
# Feat:  Added all archive functions
## Added :
- Add to archive
- delete from the archive
- restore form archive

## Ignore :
 - NoteServices Directory (Moved all related functionality files into a single directory)

### CODE REVIEW

**Review by harshitpaliwal95 — APPROVED** [review]
code looks good to me
- Ui and the app is really good
- keep up the work

File: src/contexts/notes-context.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { noteReducer } from "../reducers/noteReducer";
 const initialState = {
   notes: [],
   editNote: false,
+  archive: [],
```
Comment (ivinayakg) [inline src/contexts/notes-context.js:7]: great that you are managing it in one state only

File: src/Pages/ArchivePage/ArchivePage.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const location = useLocation();
+  const {
+    authState: { encodedToken },
+  } = useAuth();
+  const {
+    noteState: { archive },
+    noteDispatch,
+  } = useNotes();
+  const unArchiveHandler = async (note) => {
+    const response = await restoreArchiveService(note, encodedToken);
```
Comment (ivinayakg) [inline src/Pages/ArchivePage/ArchivePage.jsx:17]: add the token in the api call only

**Review by ivinayakg — APPROVED** [review]

File: src/Pages/ArchivePage/ArchivePage.jsx (hunk 2 above)
Comment (dusariudaykumar) [reply src/Pages/ArchivePage/ArchivePage.jsx:17]: sure, I will add this

**Pull request merged**